esphome/components/sps30/sps30.cpp

```diff
@@ -132,6 +138,30 @@ void SPS30Component::update() {
     }
     return;
   }
+
+  // If its not time to take an action, do nothing.
+  if (millis() < this->next_state_ms_) {
```

Comment:
```suggestion
  if ((int32_t) (this->next_state_ms_ - millis()) > 0) {
```

If `next_state_ms_` rolls over to say 0, this will not block like it is supposed to. Need to do a subtract here. Rolling counters can never be compared without a subtraction to account for the rollover.